module_bom.go

```diff
@@ -94,6 +103,31 @@ func (m ModuleBOM) Generate(workingDir string) ([]packit.BOMEntry, error) {
 				Algorithm: algorithm,
 				Hash:      entry.Hashes[0].Content,
 			}
+		} else {
+			// if the cyclonedx-bom tool doesn't find a hash (Node Engine v15+)
+			// look up the integrity field for the package from the `package-lock.json`
+			if packageLock.Name == "" {
+				packageLock, err = getLockFile(workingDir)
+			}
+
+			// if the package-lock.json is not retrievable, do not error out
+			// just skip trying to get checksums
+			if err == nil {
+				alg, hash, err := retrieveIntegrityFromLockfile(packageLock, entry.Name)
+				if err != nil {
+					return nil, err
+				}
+
+				algorithm, err := packit.GetBOMChecksumAlgorithm(alg)
+				if err != nil {
+					return nil, err
+				}
+
+				packitEntry.Metadata.Checksum = packit.BOMChecksum{
+					Algorithm: algorithm,
+					Hash:      hash,
+				}
+			}
```

Comment:
We handle the `err == nil` case here, but what happens when `getLockFile` returns an error? Nothing? Is that okay? Could we be more intentional about that cases it is okay to see an error in?

Reply:
I thought about this, because I realize that it's really weird. If it fails here it will be one of two reasons:

1. there is no `package-lock.json` which is definitely plausible. Not all apps will have it, and we do not care if that's the case. No need to error out here. I thought about doing an `os.Stat` for this case, but it just doesn't really seem necessary given we don't care if it doesn't exist.

2. The `package-lock.json` cannot be unmarshalled successfully for us to grab the `integrity`. In this case, I also don't think we really care. If this errors out I do not think it's a cause for the whole build phase to error, but more for us to just not try to include hashes and move on.